lib/Doctrine/Collection/Iterator.php

```diff
@@ -74,6 +74,7 @@ public function __construct($collection)
      *
      * @return void
      */
+    #[\ReturnTypeWillChange]
```

Comment:
IIRC we moved to implement the "right" return type for the Iterator interface in Symfony1

We could do that, instead of the `ReturnTypeWillChange`.

WDYT?

Reply:
If so, I fear we have mixed it. Here are some examples with just `#[\ReturnTypeWillChange]` attributes:

* https://github.com/FriendsOfSymfony1/symfony1/blob/master/lib/form/sfForm.class.php#L1167
* https://github.com/FriendsOfSymfony1/symfony1/blob/master/lib/form/sfFormFieldSchema.class.php#L157
* https://github.com/FriendsOfSymfony1/symfony1/blob/master/lib/event/sfEvent.class.php#L146
* https://github.com/FriendsOfSymfony1/symfony1/blob/master/lib/request/sfRequest.class.php#L222

Don't get me wrong, I'm happy to change it, but this pr does not cover all return type occurrences in the codebase and I'd rather apply this to add just the missing ones in the same way the others have been fixed.

Another pr could change all of the currently fixable changes, but as long as we are not going to >= php 8, it's not possible to get rid of all of them.

Reply:
Another thought: Changing the signature might lead to a BC change, as existing subclasses must be changed to reflect the new signature.